Lib/datetime.py

```diff
@@ -2115,6 +2138,10 @@ def __reduce__(self):
 datetime.resolution = timedelta(microseconds=1)
 
 
+def _iso_long_year_helper(year):
+    return (year + (year // 4) - (year // 100) + (year // 400)) % 7
```

Comment:
TODO: Definitely need a comment explaining what this function does and means. Possibly define it *in* `fromisocalendar` instead of here.